server/routes/components.ts

```diff
@@ -90,11 +90,14 @@ export default function routes({ serviceCatalogueService, redisService }: Servic
 
     const component = await serviceCatalogueService.getComponent(componentName)
     const environments = component.environments?.filter(environment => environment.name === environmentName)
+    const activeAgencies =
+      environments.length === 0 ? '' : formatActiveAgencies(environments[0].active_agencies as Array<string>)
 
```

Comment:
super minor.. 

It seems like there is an assumption that there will always be a single environment on line 99. `environment: environments[0],`. So you could extract out an `environment` variable earlier and not have the extra `environments.length === 0` check. 

not sure the reason for the type assertion for `as Array<String>` but that could be that the type on  `formatActiveAgencies` doesn't support passing `undefined`, if so changing the parameter to `activeAgencies: Array<string> | undefined` might mean you don't need that.

Reply:
> It seems like there is an assumption that there will always be a single environment on line 99. environment: environments[0],. So you could extract out an environment variable earlier and not have the extra environments.length === 0 check.
I think there is an assumption that there will be at least one environment set up for a component, so I guess this may not be the case. I'll take a look in a future pass.

> not sure the reason for the type assertion for as Array<String> but that could be that the type on formatActiveAgencies doesn't support passing undefined, if so changing the parameter declaration on the function to activeAgencies: Array<string> | undefined might mean you don't need that.

ah yes, thanks for this, makes sense. The type returned from strapi is an undefined json object, so the above makes sense. Again i'll update in the next pass through.